code/nrf-connect/prstlib/boards/arm/bparasite_nrf52833/bparasite_nrf52833.dts

```diff
@@ -0,0 +1,172 @@
+/dts-v1/;
+#include <nordic/nrf52833_qiaa.dtsi>
+#include "bparasite_nrf52833-pinctrl.dtsi"
+
+/ {
+	model = "A soil moisture sensor based on Nordic's nRF52833";
+	compatible = "rbaron,bparasite_nrf52833";
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+
+	zephyr,user {
+		io-channels = <&adc 0>, <&adc 1>, <&adc 2>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0: led_0 {
+			// P0.28.
+			gpios = <&gpio0 28 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			// P0.12.
+			gpios = <&gpio0 12 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+			label = "Push button switch 0";
+		};
+	};
```

Comment:
Maybe I'm missing something, but I think none of the board revisions so far  have buttons.  `nrf52840dk_nrf52840.overlay`? or maybe it's for a future board revision?

Reply:
Good eye. I am working on a [v2 hardware review](https://github.com/rbaron/b-parasite/commits/hardware-v2) with revised soil sensing circuitry, better RF module footprint for hand soldering & a couple of buttons. It's mostly ready.

I will move the devicetree button definition to a 2.0.0 board revision once it arrives. It's just there because I already wanted to try the input setup in the ncs.

Reply:
Cool! Please let me know if I can pitch in for a few samples in a group buy or order. Buttons open up a lot of possibilities.

Reply:
Will do. I would love to get this to a place where it becomes a nice dev board with samples for different protocols, not just BLE. I think this PR is a nice step in that direction.

Reply:
Took a look at the `hardware-v2` branch. I like the buttons! Do you plan on opening an issue/PR for comments before you set it in stone? It's good as it is, but, since you mention "dev board", I have a couple of wishes :pleading_face:

Reply:
I will open a PR soon! Just want to merge this PR first. But I would very much like to keep it simple and cheap and small. What do you have in mind?

Reply:
Just ideas at this point, possibly far-fetched or not really useful, but here they are:

Footprint for USB connector, optionally LDO power from USB on the back side. I realize those will take up some real estate, but it should be possible to place the connector opposite to the programming header, unpopulated by default.  This will make the board usable without the extra SWD hardware. There's a bootloader for the chip by Adafruit. If not USB connector + ldo, maybe just test pads for DP/DN and GND/3V0 to allow for a jig which connects to them.

And, this may be a bit too much, but how about TH instead of test points for the header? This will make it usable without the pogo-pin clamp. BTW, I built myself a jig because I didn't have a clamp, but had bare spring-loaded pins handy:

![IMG_20221206_224730332_HDR](https://user-images.githubusercontent.com/98131/206020538-a8595294-a1d1-485c-9d4e-90389da73fb0.jpg)

I was able to import the board from KiCAD into FreeCAD and build around it.  :heart: opensource :heart: 

But I wish I could just plug in some 2.45mm-pitch jumper wires. Clamps should still be usable with a TH footprint and it looks like holes don't complicate routing.

Nice touch with the silkscreen too! However, the C = T/R equation would make my 4h grade math teacher frown. She couldn't stand seeing a `=` sign not aligned with a fractional line. This should be a trivial fix :)

Reply:
Adding a USB connector and LDO are a not very much in the plans, at least for v2. I want to keep it cheap and low profile, and you'd still need the SWD for the initial flash with the USB bootloader I think.

Reply:
All true, but having the footprints will not add any cost, one would populate them only if needed. Anyway, I agree the utility of this is questionable if the board is destined to live stuck in a pot.

Reply:
@kzyapkov what kind of printer/sheet label/software did you use to print those labels? Was it [blabel](https://github.com/Edinburgh-Genome-Foundry/blabel)?

Reply:
You should ask @oleo65, I was lucky enough to  snatch a few from his batch -- they came with the labels.

Reply:
Tks, I managed to whip something together with blabel. Cool idea @oleo65 👌

Reply:
I had a more or less sophisticated python script lying around for generating small QR code labels. I initially used it for creating random ID labels but since the MAC address was not too long, I was able to reuse it.

It is customized for some off the shelve A4 sheet labels.

Most standard hardware label printer don't have a good enough resolution to print so small QR codes which are still readable I suppose. 🤷